libraries/aws/shadow/src/aws_iot_shadow_api.c

```diff
@@ -349,7 +349,7 @@ static AwsIotShadowError_t _setCallbackCommon( IotMqttConnection_t mqttConnectio
     }
 
     /* Check parameters. */
-    status = _validateThingNameFlags( ( _shadowOperationType_t ) ( type + SHADOW_OPERATION_COUNT ),
+    status = _validateThingNameFlags( ( _shadowOperationType_t ) ( ( int32_t ) type + SHADOW_OPERATION_COUNT ),
```

Comment:
Is your intent to typecast "type" and then let the Integer promotion take care of the rest, or is your intent to cast the result of the opertation? I think brackets make the intent clearer in these cases.

Reply:
The specific violation is that we are doing arithmetic with an enum (`type`).

Reply:
Then I expect (( int32_t) type ) + SHADOW_OPERATION_COUNT